packages/flutter/lib/src/material_animated_icons/animated_icons.dart

```diff
@@ -79,9 +80,166 @@ class AnimatedIcon extends StatelessWidget {
   /// specified, either directly using this property or using [Directionality].
   final TextDirection textDirection;
 
+  static final _UiPathFactory _pathFactory = () => new ui.Path();
+
   @override
   Widget build(BuildContext context) {
-    // TODO(amirh): implement this.
-    return new Container();
+    // TODO(amirh): implement semantics, text direction, scaling.
+    final _AnimatedIconData iconData = icon;
+    return new CustomPaint(
+      size: iconData.size,
+      painter: new _AnimatedIconPainter(
+        iconData.paths,
+        progress,
+        color,
+        _pathFactory,
+      ),
+    );
+  }
+}
+
+typedef ui.Path _UiPathFactory();
+
+class _AnimatedIconPainter extends CustomPainter {
+  _AnimatedIconPainter(
+    this.paths,
+    this.progress,
+    this.color,
+    this.uiPathFactory,
+  ) : super(repaint: progress);
+
+  // This list is assumed to be immutable, changes to the contents of the list
+  // will not trigger a redraw as shouldRepaint will keep returning false.
+  final List<_PathFrames> paths;
+  final Animation<double> progress;
+  final Color color;
+  final _UiPathFactory uiPathFactory;
+
+  @override
+  void paint(ui.Canvas canvas, Size size) {
+    for (_PathFrames path in paths)
+      path.paint(canvas, color, uiPathFactory, progress.value);
```

Comment:
you could save a few cycles by clamping the value at the top of this method and just passing the clamped value to all these paint methods.